tests/e2e/specs/users.test.js

```diff
@@ -0,0 +1,192 @@
+		).toContain( 'testuser' );
+	} );
+
+	it( 'Should return No users found. when searching for a user that does not exist', async () => {
+		// Wait for the search field to appear and focus it
+		const userSearchInput = await page.waitForSelector( '#user-search-input' );
+		userSearchInput.focus();
+
+		// Type the new username in the search input
+		await page.keyboard.type( 'nonexistinguser' );
+
+		// Move to the search button and click on it
+		await page.keyboard.press( 'Tab' );
+		await page.keyboard.press( 'Enter' );
+		await page.waitForNavigation();
+
+		// Check that there is only one user row
+		const allUsersRows = await page.$$( '#the-list tr.no-items' );
+		expect( allUsersRows.length ).toBe( 1 );
+		
+		// Check that the remaining row contains "No users found."
+		const notFoundUserRow = await page.waitForSelector( '#the-list tr.no-items' );
+		expect(
+			await notFoundUserRow.evaluate( ( element ) => element.innerText )
+		).toContain( 'No users found.' );
+	} );
+
+	it( 'Correctly edit a user first and last names', async () => {
+		await goToUserProfilePage( "testuser" );
+
+		// Wait for the user first name input to appears
+		await page.waitForSelector( 'input#first_name' );
+		
+		// Focus on the user first name input field
+		await page.focus( 'input#first_name' );
+
+		// Edit the user first and last names
+		await page.keyboard.type( 'Test' );
+		await page.keyboard.press( 'Tab' );
+		await page.keyboard.type( 'User' );
+
+		// Focus on the submit button and save the changes
+		await page.focus( 'input#submit' );
+		await page.keyboard.press( 'Enter' );
+
+		// Wait for the success notice message to show
+		await page.waitForSelector( '#message' );
+
+		// Go back to the users list page
+		await visitAdminPage( 'users.php' );
+
+		// Check that the new user complete name is "Test User"
+		const editedUserFullName = await page.$x(
+			`//td[contains( @class, "column-name" )][contains( text(), "Test User" )]`
+		);
+		expect( editedUserFullName.length ).toBe( 1 );
+	} );
+
+	it( 'Correctly changes the role of a user', async () => {
+		await goToUserProfilePage( "testuser" );
+
+		// Wait for the role field to appears
+		await page.waitForSelector( 'select#role' );
+
+		// Change the user role to author
+		await page.select( 'select#role', 'author' );
+
+		// Focus on the submit button and save the changes
+		await page.focus( 'input#submit' );
+		await page.keyboard.press( 'Enter' );
+
+		// Wait for the success notice message to show
+		await page.waitForSelector( '#message' );
+
+		// Go back to the users list page
+		await visitAdminPage( 'users.php' );
+
+		// Check that the new user role name is "author"
+		const editedUserRole = await page.$x(
+			`//td[contains( @class, "column-role" )][contains( text(), "Author" )]`
+		);
+		expect( editedUserRole.length ).toBe( 1 );
+	} );
+
+	it( 'Should not allows the main admin user to change their role', async () => {
+		await goToUserProfilePage( "admin" );
```

Comment:
Again, we can't know if the current user is `admin` or has some other name. I think the most reliable way to do this would be to not create any new user before this test, so we know there is only one user - the current one - on the list.

(Of course doing this will mean removing `createBasicUser` from `beforeEach` and adding it inside each of the other tests, but there's one other test that also doesn't need it: the non-existing user one. So I guess it could be worse 😅 )

I'm also thinking that instead of checking the user's profile page, we can use the "Change role to..." dropdown to try to change their role to, say, "Subscriber". Then check that an error message displays at the top of the page, and the role wasn't changed.